src/windows/find_tools.rs

```diff
@@ -632,11 +632,15 @@ mod impl_ {
 
     // Inspired from official microsoft/vswhere ParseVersionString
     // i.e. at most four u16 numbers separated by '.'
-    fn parse_version(version: &str) -> Option<Vec<u16>> {
-        version
-            .split('.')
-            .map(|chunk| u16::from_str(chunk).ok())
-            .collect()
+    fn parse_version(version: &str) -> Option<[Option<u16>; 4]> {
+        let mut iter = version.split('.').map(|chunk| u16::from_str(chunk)).fuse();
+        let mut get_next_number = move || iter.next().transpose().ok();
+        Some([
+            get_next_number()?,
+            get_next_number()?,
+            get_next_number()?,
+            get_next_number()?,
+        ])
```

Comment:
Hm, could this just return `Option<[u16; 4]>` (the array is then the same size as a `u64`)? Using zero for the missing parts is fine.